boto/https_connection.py

```diff
@@ -77,7 +77,7 @@ def ValidateCertificateHostname(cert, hostname):
         "validating server certificate: hostname=%s, certificate hosts=%s",
         hostname, hosts)
     for host in hosts:
-        host_re = host.replace('.', '\.').replace('*', '[^.]*')
+        host_re = host.replace('.', r'\.').replace('*', '[^.]*')
```

Comment:
So only one of these parameters are regexes, or is it because the `\` causes problems but not the other ones ?

Reply:
yes as `.` is not a valid backslash character it raises warnings. making it a raw string will solve this warning as warnings are excluded from raw strings.

Reply:
Same comment as for other such cases.

Reply:
done